src/main/daml/Daml/Finance/Holding/Util.daml

```diff
@@ -7,31 +7,57 @@ module Daml.Finance.Holding.Util
   , transferImpl
   ) where
 
-import DA.Set qualified as S (delete, insert, isSubsetOf, null, singleton)
-import Daml.Finance.Interface.Account.Account qualified as Account (CreditAndDebit(..), I, R, exerciseInterfaceByKey)
+import DA.Action (foldlA)
+import DA.Set qualified as S (delete, insert, isSubsetOf, null, singleton, toList)
+import Daml.Finance.Interface.Account.Account qualified as Account (Credit(..), Debit(..), I, R, exerciseInterfaceByKey)
 import Daml.Finance.Interface.Account.Util (fetchAccount)
 import Daml.Finance.Interface.Holding.Base (getLockers)
 import Daml.Finance.Interface.Holding.Base qualified as Base (Acquire(..), I, Lock(..), LockType(..), Release(..))
 import Daml.Finance.Interface.Holding.Transferable qualified as Transferable (I, Transfer(..))
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as Instrument (qty)
 import Daml.Finance.Interface.Util.Common (fetchInterfaceByKey)
 import Prelude hiding (null)
 
 -- | Implementation of transfer for the Transferable interface.
 transferImpl : ContractId Transferable.I -> Transferable.Transfer
   -> Update (ContractId Transferable.I)
 transferImpl self Transferable.Transfer{actors; newOwnerAccount} = do
-  transferable <- fetch self
-  vAccount <- view <$> fetchAccount transferable
+  let baseCid = toInterfaceContractId @Base.I self
+  base <- fetch baseCid
+  account <- fetchAccount base
+  let
+    vBase = view base
+    vAccount = view account
+    lockers = getLockers base
   vNewAccount <- view <$> fetchInterfaceByKey @Account.R @Account.I newOwnerAccount
```

Comment:
We can porobably use `fetchAccount` here as well

Reply:
`fetchAccount : (HasToInterface t Base.I) => t -> Update Account.I` takes a holding not an account key